### PR TITLE
Unified UTS'graphServer metaServer storageServer, Delete checkSpaceExist(), Delete FileBasedSchemaManager, Move AdHocSchemaManager to storage/test

### PR DESCRIPTION
**Pull request by laura-ding**
1、Delete checkSpaceExist()
2、Delete FileBasedSchemaManager
3、Move AdHocSchemaManager to storage/test
4、Unified UTS'graphServer metaServer storageServer

### CODE REVIEW

**Comment by laura-ding** [comment]
Jenkins go

File: src/common/test/ServerContext.h [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    std::unique_ptr<apache::thrift::ThriftServer>      server_{nullptr};
+    std::unique_ptr<thread::NamedThread>               thread_{nullptr};
+    // To keep meta and storage's KVStore
+    std::unique_ptr<kvstore::KVStore>                  KVStore_{nullptr};
+    uint16_t                                           port_{0};
+};
+
+static void mockCommon(test::ServerContext *sc,
+                       std::string name,
+                       uint16_t port,
```
Comment (boshengchen) [inline src/common/test/ServerContext.h]: const std::string &name

File: src/meta/ServerBasedSchemaManager.cpp [hunk 2]
```diff
@@ -40,8 +41,12 @@ std::shared_ptr<const SchemaProviderIf>
...
                                        folly::StringPiece tagName,
                                        SchemaVer ver) {
-    auto space = toGraphSpaceID(spaceName);
-    return getTagSchema(space, toTagID(space, tagName), ver);
+    auto spaceStatus = toGraphSpaceID(spaceName);
+    if (!spaceStatus.ok()) {
+        return std::shared_ptr<const SchemaProviderIf>();
+    }
+    auto spaceId = spaceStatus.value();
+    return getTagSchema(spaceId, toTagID(spaceId, tagName), ver);
```
Comment (boshengchen) [inline src/meta/ServerBasedSchemaManager.cpp]: Verify that the return value of `toTagID`  is better? It might return -1 with no error.

Comment (laura-ding) [reply src/meta/ServerBasedSchemaManager.cpp]: Usually, the cache can be found without having to judge the toTagID() every time. Sometimes we can't find it, we just do an extra search.

File: src/common/test/ServerContext.h [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    std::unique_ptr<apache::thrift::ThriftServer>      server_{nullptr};
+    std::unique_ptr<thread::NamedThread>               thread_{nullptr};
+    // To keep meta and storage's KVStore
+    std::unique_ptr<kvstore::KVStore>                  KVStore_{nullptr};
+    uint16_t                                           port_{0};
+};
+
+static void mockCommon(test::ServerContext *sc,
```
Comment (dangleptr) [inline src/common/test/ServerContext.h]: Well done! 👍

File: src/meta/SchemaManager.h [hunk 4]
```diff
@@ -49,89 +49,18 @@ class SchemaManager {
     virtual SchemaVer getNewestEdgeSchemaVer(folly::StringPiece spaceName,
                                              folly::StringPiece typeName) = 0;
 
-    virtual GraphSpaceID toGraphSpaceID(folly::StringPiece spaceName) = 0;
+    virtual StatusOr<GraphSpaceID> toGraphSpaceID(folly::StringPiece spaceName) = 0;
 
     virtual TagID toTagID(GraphSpaceID space, folly::StringPiece tagName) = 0;
```
Comment (dangleptr) [inline src/meta/SchemaManager.h]: toTagID  and  toEdgeType  needs to return StatusOr too.

Comment (laura-ding) [reply src/meta/SchemaManager.h]: It's used within GoExecutor, and GoExecutor doesn't do the exception handling. When it completes the exception handling process, modify it.

Comment (dangleptr) [reply src/meta/SchemaManager.h]: That's ok.  Change the interface firstly.  Other modules need to depend on it

**Comment by dangleptr** [comment]
Close #342

**Comment by laura-ding** [comment]
jenkins go

**Comment by dutor** [comment]
Jenkins go!

**Pull request merged**